### PR TITLE
Add default/fallback behavior for json.dumps

### LINKED ISSUE #41: structlog integration - TypeError when passing object
Right now when you add an object to your log like this:

```python
            logger.info(
                "request_started",
                request=request,
                user_agent=request.META.get("HTTP_USER_AGENT"),
            )
```

You end up with `TypeError` like this: `<WSGIRequest: GET '/healthcheck/'> is not JSON serializable` in `json_dumps` (`ecs_logging/_utils.py`):

```python
        if value:
            return "{%s,%s" % (
                ordered_json,
                json.dumps(value, sort_keys=True, separators=(",", ":"))[1:],
            )
```

You should use a `default` argument for `json.dumps` and provide fallback mechanism to handle this situation. See https://www.structlog.org/en/stable/_modules/structlog/processors.html#JSONRenderer and `_json_fallback_handler`.

### PR DESCRIPTION
**Pull request by basepi**
Fixes #41

### CODE REVIEW

File: ecs_logging/_utils.py [hunk 1]
```diff
@@ -151,17 +151,46 @@ def json_dumps(value):
         # case the given values aren't strings (even though
         # they should be according to the spec)
         ordered_json = ",".join(
-            '"%s":%s' % (k, json.dumps(v, sort_keys=True, separators=(",", ":")))
+            '"%s":%s'
+            % (
+                k,
+                json.dumps(
```
Comment (sethmlarson) [inline ecs_logging/_utils.py:157]: Starting to be a lot of duplication in this function, maybe we need to wrap `json.dumps(..., <all-of-our-options>)` somewhere? Can be an inline function or `functools.partial`?

**Review by sethmlarson — CHANGES_REQUESTED** [review]
I'm still kinda "eh" about whether we should simply smooth over errors where the user isn't giving us something JSON serializable? Isn't an error better in this case instead of polluting their logs cluster with object reprs?

File: ecs_logging/_utils.py (hunk 1 above)
Comment (basepi) [reply ecs_logging/_utils.py]: Done.

**Comment by basepi** [comment]
@sethmlarson Obviously at least one person has run into this, and expects us to not fail but rather deal with the object. Doing it this way allows a user to add the `__structlog__()` method to give it a good log representation. The `__repr__()` backup will make it pretty obvious what's going on as well IMO.

**Review by sethmlarson — APPROVED** [review]
LGTM

**Review by beniwohli — APPROVED** [review]
LGTM!

**Comment by basepi** [comment]
Just waiting on #46 which will turn these tests green.

**Review by sethmlarson — APPROVED** [review]
`functools.partial` change looks good to me!

**Pull request merged**